.github/workflows/publish_charm.yaml

```diff
@@ -19,12 +22,22 @@ on:
         required: false
         description: |
           Tag prefix, useful when bundling multiple charms in the same repo.
+      charmcraft-channel:
+        description: Charmcraft channel to use for the integration test
+        type: string
+        default: latest/stable
 
 env:
   REGISTRY: ghcr.io
   OWNER: ${{ github.repository_owner }}
 
 jobs:
+  vars:
+    runs-on: ubuntu-22.04
+    outputs:
+      charm_working_directory: ${{ inputs.charm-working-directory || inputs.working-directory }}
+    steps:
+      - run: echo "Exposing env vars"
```

Comment:
Why not add `charm_working_directory` as a workflow env (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#env) so its available to all jobs without the need for a new job (requesting a runner which counts against our organisational quota)?

Reply:
mmm I think I read that context cannot be used in defaults.run (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrun). Let me try...

Reply:
https://github.com/canonical/operator-workflows/actions/runs/8571566147

Reply:
I will try in the step.working-directory

Reply:
Also for `uses`.... https://github.com/canonical/operator-workflows/actions/runs/8572352463 I will go to the original solution with outputs

Reply:
Ok, thanks for trying. Its a bit strange, as https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability indicates that its possible to use env in `defaults.run`, although https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#defaultsrun actually states that its not possible to use contexts.